File: 01. Linked List/1. Singly-Linked-List/InterviewBit Questions/11. Rotate LL by K place/solution 11.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     public int val;
 *     public ListNode next;
 *     ListNode(int x) { val = x; next = null; }
 * }
 */
public class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        ListNode last=head;
        ListNode t=head;
        int size=0;
        while(t.next!=null)
        {
            size++;
            t=t.next;
        }
        size++;
        
        if(size==1){
            return head;
        }
        if(k>size){
            k=k%size;
        }
        if(k==0) return head;
        
        
        int lpos=size-k;
        
        while(lpos>1){
            last=last.next;
            lpos--;
        }
        
        ListNode start=last.next;
        
        t.next=head;
        last.next=null;
        
        return start;
    }
}

File: 01. DataStructures/01. Linked List/1. Singly-Linked-List/InterviewBit Questions/11. Rotate LL by K place/solution_11.py
class Solution:

    def rotateRight(self, A, B):
        aa = []
        if A == None:
            return A
        while(A):
            aa.append(A.val)
            A = A.next
        bb = []
        bc = []
        B = (B % len(aa))
        i = len(aa)-B
        while(i < len(aa)):
            bb.append(aa[i])
            i += 1
        i = 0
        while(i < len(aa)-B):
            bb.append(aa[i])
            i += 1
        head = ListNode(bb[0])
        cur = head
        for i in range(1, len(bb)):
            cur.next = ListNode(bb[i])
            cur = cur.next
        return head
